Source/Core/Check.js

```diff
@@ -49,31 +49,35 @@ define([
      * Throws if test is greater than maximum
      *
      * @param {Number} test The value to test
+     * @param {String} name The name of the variable being tested
      * @param {Number} maximum The maximum allowed value
      * @exception {DeveloperError} test must not be greater than maximum
      * @exception {DeveloperError} Both test and maximum must be typeof 'number'
      */
-    Check.numeric.maximum = function (test, maximum) {
-        Check.typeOf.number(test);
-        Check.typeOf.number(maximum);
+    Check.numeric.maximum = function (test, name, maximum) {
+        Check.typeOf.number(test, 'test');
+        Check.typeOf.string(name, 'name');
```

Comment:
It's more helpful here than in the `typeOf` checks since there is an additional parameter to potentially misplace here, and the old interface had different parameters. I guess it's probably most line with the fail-early strategy to type check all parameters, as long as there's a chance to catch a problem early and as long as we can afford it. What do you think?